src/distributions/mod.rs

```diff
@@ -189,9 +189,29 @@ impl<'a, T, D: Distribution<T>> Distribution<T> for &'a D {
 /// let mut rng = thread_rng();
 /// let mut erased_rng: &mut RngCore = &mut rng;
 /// let val: f32 = erased_rng.sample(Uniform);
-/// println!("f32 from [0,1): {}", val);
+/// println!("f32 from (0,1): {}", val);
 /// ```
 ///
+/// # Open interval for floats
+/// In theory it is possible to choose between an open interval `(0, 1)`, and
+/// the half-open intervals `[0, 1)` and `(0, 1]`. All can give a distribution
+/// with perfectly uniform intervals. Many libraries in other programming
+/// languages default to the closed-open interval `[0, 1)`. We choose here to go
+/// with *open*, with the arguments:
+///
+/// - The chance to generate a specific value, like exactly 0.0, is *tiny*. No
+///   (or almost no) sensible code relies on an exact floating-point value to be
+///   generated with a tiny chance (1 in 2^23 for `f32`, and 1 in 2^52 for
```

Comment:
1/2^23 is not really tiny, it corresponds to just 32 MiB of random data. I could be relevant to generate zero for fuzzing purposes.

Reply:
Hmm, maybe better to rewords this a bit. Something like "1 in ~8 million (2^23) for `f32`". Nor sure if that counts as tiny, I had mostly `f64` in mind when writing that :smile:.

> It could be relevant to generate zero for fuzzing purposes.

Good point. Of course there are always some valid uses for `[0, 1)`. In this case, fuzzing, we concluded in https://github.com/dhardy/rand/issues/83 that you are better of using some specific library and trait like `Arbitrary` instead of the old `Rand` trait or our distributions.

Reply:
That point was more about converting to user types. 

The chance of never getting 0 from 1 sample of n=2^23 choices is (n-1)/n or approx. 0.9999998807907104.

The chance of never getting 0 from k samples of n choices is ((n-1)/n)^k; if we take k=2^23 samples that translates to a probability of 37% of never sampling 0, so you might want to up your sample size a bit:
```
>>> ((2**23 - 1.0) / (2**23))
0.9999998807907104
>>> ((2**23 - 1.0) / (2**23))**(2**23)
0.3678794192441178
>>> ((2**23 - 1.0) / (2**23))**(2**24)
0.13533526710338942
>>> ((2**23 - 1.0) / (2**23))**(2**25)
0.018315634521945755
```
This is assuming the f64 ops in Python are able to represent these computations with sufficient accuracy; I think they should be able to.